src/useValidator/__docs__/example.stories.tsx

```diff
@@ -0,0 +1,48 @@
+import * as React from 'react';
+import { useState } from 'react';
+import { IValidatorDeferred, useDebounceCallback, useValidator } from '../..';
+
+export const ExampleStories: React.FC = () => {
+  const [text, setText] = useState('');
+
+  // as deferred validator is unable to infer the type of validity
+  // state - we should define it ourself
+  type ITextValidityState = { isValid: boolean | undefined; error: Error | undefined };
+
+  // debounced callback is deferred callback so we should use deferred type
+  // of validator (the one that receives dispatcher as an argument)
+  const validator = useDebounceCallback<IValidatorDeferred<ITextValidityState>>(
+    (d) => {
+      const isValid = !text.length || text.length % 2 === 1;
+
+      d({
+        isValid,
+        error: !isValid ? new Error('text length should be an odd length') : undefined,
+      });
+    },
+    150,
+    [text]
+  );
+
+  // validity state type if inferred from validator
+  const [validity] = useValidator(validator, [validator]);
+
+  return (
+    <div>
+      <div>
+        In case below input has an odd number of characters - it is valid, otherwise, the error
```

Comment:
This isn't exactly clear. Let's reword it to something more like "The input below is only valid if it has an odd number of characters".